touchforms/backend/xformplayer.py

```diff
@@ -718,9 +720,17 @@ def current_question(xform_session, override_state=None):
 
 
 def heartbeat(session_id):
-    # just touch the session
-    with global_state.get_session(session_id) as xfsess:
-        return {}
+    """
+    The heartbeat will update the session's last seen timestamp and sequence ID.
+    """
+    # todo: cz as of 8/5/2015 I'm pretty sure this isn't necessary. persistence can
+    # just pop this back into the context whenever necessary so no longer super relevant
+    # to keep around in memory while the tab is open.
+    with global_state.get_lock(session_id):
+        # this updates the session sequence ID and timestamp and persists it
+        with global_state.get_session(session_id):
```

Comment:
hmm, these are the only two lines i see with `heartbeat` in them in your log [doc](https://docs.google.com/document/d/1IwQCOaqZA9ptUmc7RrECzr1zjzPgYfAbvbC52lxAo4A/edit)

```
INFO 2015-07-30 12:28:01,858 xformserver 3093 8687 Received action heartbeat for session <unknown>
INFO 2015-07-30 12:28:01,858 xformserver 3093 8687 Finished processing action heartbeat in 0.999927520752 ms for session <unknown> in domain '<unknown>'
```